### PR TITLE
Rating chart updates & stat fixes

### LINKED ISSUE #39: DTO: Incorporate decay into user rating graph
Currently, the rating graph only shows specifics about matches, and how **matches** affected the decay. However, decay is not tied to a match at all. It is a foreign adjustment. Therefore, we may opt to do one of two things:

- Have optional nullable for all props that decay doesn't touch (matchid, etc.) in the returned DTO
- Return a dynamic object

Currently leaning more towards option 1 because this is probably just better for maintainability.

### PR DESCRIPTION
**Pull request by hburn7**
Closes #39 

@AkinariHex There are breaking changes.

`/api/stats/{playerId}` & `/api/stats/{username}`:
- There is a new field `ratingChart` (replaces `ratingStats`). Each item is still grouped into the same "1 collection per day" format.
- The rating chart now shows whether the item is an adjustment. Any item with `isAdjustment = true` means the rating change comes from something other than an osu! match
- Removed `mostPlayedTeammateName`, `mostPlayedOpponentName`, and `bestTeammateName` from `matchStats` field.

```json
  "ratingChart": {
    "chartData": [
      [
        {
          "name": "S1AT: (Peanut Overcharge) vs (average osu enjoyers)",
          "matchId": 37546,
          "matchOsuId": 96891231,
          "matchCost": 1.4739258926651206,
          "ratingBefore": 699.824077356349,
          "ratingAfter": 749.7960149715763,
          "volatilityBefore": 225,
          "volatilityAfter": 224.30826246700434,
          "ratingChange": 49.97193761522726,
          "volatilityChange": -0.6917375329956599,
          "isAdjustment": false
        }
      ],
      [
        {
          "name": "S1AT: (osu!zhdayu) vs (Peanut Overcharge)",
          "matchId": 37566,
          "matchOsuId": 97194288,
          "matchCost": 1.465160329925816,
          "ratingBefore": 749.7960149715763,
          "ratingAfter": 799.114430922645,
          "volatilityBefore": 224.30826246700434,
          "volatilityAfter": 223.57843078967647,
          "ratingChange": 49.3184159510688,
          "volatilityChange": -0.7298316773278657,
          "isAdjustment": false
        }
      ],
```

Here is an example of decay:
```json
[
        {
          "name": "Decay",
          "matchId": null,
          "matchOsuId": null,
          "matchCost": null,
          "ratingBefore": 1014.5539768123309,
          "ratingAfter": 1011.8539768123309,
          "volatilityBefore": 188.5423886396073,
          "volatilityAfter": 188.97151191099871,
          "ratingChange": -2.7000000000000455,
          "volatilityChange": 0.42912327139140416,
          "isAdjustment": true
        }
      ],
      [
        {
          "name": "Decay",
          "matchId": null,
          "matchOsuId": null,
          "matchCost": null,
          "ratingBefore": 1011.8539768123309,
          "ratingAfter": 1009.1539768123308,
          "volatilityBefore": 188.97151191099871,
          "volatilityAfter": 189.39966291925845,
          "ratingChange": -2.7000000000000455,
          "volatilityChange": 0.4281510082597322,
          "isAdjustment": true
        }
      ],
```

### CODE REVIEW

**Comment by AkinariHex** [comment]
I'm fine with the changes and works with the frontend after few tweeks.

**Review by AkinariHex — APPROVED** [review]
I'm fine with the changes and works with the frontend after few tweeks.

**Comment by hburn7** [comment]
Ready for review @bugg86

File: API/Services/Implementations/PlayerStatsService.cs [hunk 1]
```diff
@@ -234,11 +235,11 @@ public async Task TruncateAsync()
...
-	private async Task<IEnumerable<IEnumerable<MatchRatingStatsDTO>>> GetRatingStatsAsync(int playerId, int mode, DateTime dateMin, DateTime dateMax)
-	{
-		var ratingStats = await _ratingStatsRepository.GetForPlayerAsync(playerId, mode, dateMin, dateMax);
-		return _mapper.Map<IEnumerable<IEnumerable<MatchRatingStatsDTO>>>(ratingStats);
-	}
+	// private async Task<IEnumerable<IEnumerable<MatchRatingStatsDTO>>> GetRatingStatsAsync(int playerId, int mode, DateTime dateMin, DateTime dateMax)
+	// {
+	// 	var ratingStats = await _ratingStatsRepository.GetForPlayerAsync(playerId, mode, dateMin, dateMax);
+	// 	return _mapper.Map<IEnumerable<IEnumerable<MatchRatingStatsDTO>>>(ratingStats);
+	// }
```
Comment (bugg86) [inline API/Services/Implementations/PlayerStatsService.cs:242]: Is there a reason why this is being left in commented out and not just removed?

**Review by bugg86 — APPROVED** [review]
The commented out function should probably be removed but otherwise code looks good!

File: API/Services/Implementations/PlayerStatsService.cs (hunk 1 above)
Comment (hburn7) [reply API/Services/Implementations/PlayerStatsService.cs]: Good eye, this was what was there before the new implementation.

**Review by bugg86 — APPROVED** [review]

**Pull request merged**